### ISSUE TITLE
[bigquery] Unparsable string_agg on window functions

### ISSUE THREAD

**Issue by mauro-lanza** [sqlfluff/sqlfluff#7475]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When linting a model that has a string_agg in a window function it fails to parse it. probably related to #7367 

### Expected Behaviour

Parse it

### Observed Behaviour

```
== [test.sql] FAIL                                                                                                                                         
L:   3 | P:   5 | AL03 | Column expression without alias. Use explicit `AS`
                       | clause. [aliasing.expression]
L:   3 | P:  31 | LT02 | Expected line break and no indent before 'over'.
                       | [layout.indent]
L:   3 | P:  32 |  PRS | Line 3, Position 32: Found unparsable section: 'over
                       | (partition by agg_column) as listin...'
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

``` sql
select
    some_key,
    string_agg(listing_column) over (partition by some_key) as listing_agg,
    sum(qty_column) over (partition by some_key) as total_quantity
from {{ ref('some_table') }}

```

### Dialect

Bigquery

### Version

Python 3.12.12
sqlfluff, version 4.0.4

### Configuration

``` ini
[sqlfluff]
templater = jinja
dialect = bigquery
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by sambloom92** [comment]
This issue seems to be resolved for sqlfluff but still affects sqfluff[rs]. Please could this issue be re-opened, or should I raise a new issue?

@alanmcruickshank

**Comment by peterbud** [comment]
This fix has been added to the main branch, but has NOT been released. Have you built yourself a private version which makes you think that the rust version is not fixed?

**Comment by sambloom92** [comment]
Yes I ran it from the latest commit hash `74da8e7eee5c214c850a7655c359c39832a64b9e` via pre-commit

**Comment by peterbud** [comment]
@keraion can you have a look at this pls?

**Comment by sambloom92** [comment]
thanks @peterbud

**Comment by keraion** [comment]
What does your pre-commit file look like? Is it also installing against the same commit? I haven't confirmed this, but if you are installing from git, it will likely need to build sqlfluffrs as a pre-built wheel would not be available.

**Comment by sambloom92** [comment]
I use the commit hash above as the rev and I add sqlfluff[rs] as an additional dependency like this:
```
additional_dependencies:
  - "dbt-bigquery"
  - "sqlfluff-templater-dbt"
  - "sqlfluff[rs]"
```

**Comment by keraion** [comment]
I've not tested this, but this should install sqlfluffrs from the same commit. Note that it likely requires the Rust toolchain to correctly install.
```yaml
additional_dependencies:
  - "dbt-bigquery"
  - "sqlfluff-templater-dbt"
  - ".[rs]"
```

Actually, that probably still won't work as it's going to be looking for `sqlfluffrs==4.0.4` still which it will retrieve the same wheel from before.

**Comment by sambloom92** [comment]
I had assumed adding `sqlfluff[rs]` would be the way to opt in to the rust version, but I realise now I was just guessing as I didn't actually see this documented anywhere. I've tried the above suggestion with ".[rs]" instead - I still get the same parsing errors. Without the additional dependency it parses correctly.

**Comment by keraion** [comment]
Then I think that it leaves us pulling against the VCS URL directly until the next version is released.

```yaml
additional_dependencies:
  - "dbt-bigquery"
  - "sqlfluff-templater-dbt"
  - "git+https://github.com/sqlfluff/sqlfluff.git@74da8e7eee5c214c850a7655c359c39832a64b9e#subdirectory=sqlfluffrs"
```

**Comment by sambloom92** [comment]
Installing from the subdirectory isn't working for me:

```
$pre-commit install-hooks
...
    Building wheels for collected packages: sqlfluff, sqlfluffrs
      Building wheel for sqlfluff (pyproject.toml): started
      Building wheel for sqlfluff (pyproject.toml): finished with status 'done'
      Created wheel for sqlfluff: filename=sqlfluff-4.0.4-py3-none-any.whl size=964925 sha256=5fd01814296506482b302e63dd7575b7ae5f05defff7f48e67a2326c336c5a08
      Stored in directory: /private/var/folders/v4/npbx738j76sd232hnyb2r8zc0000gn/T/pip-ephem-wheel-cache-_18yqsw4/wheels/5c/ee/fc/8c21b8ac1b20f38fd08e95d4ff76a9fde50683e8c1e4cffb5c
      Building wheel for sqlfluffrs (pyproject.toml): started
      Building wheel for sqlfluffrs (pyproject.toml): finished with status 'error'
    Successfully built sqlfluff
    Failed to build sqlfluffrs
stderr:
      Running command git clone --filter=blob:none --quiet https://github.com/sqlfluff/sqlfluff.git /private/var/folders/v4/npbx738j76sd232hnyb2r8zc0000gn/T/pip-req-build-6kvf9w3w
      Running command git rev-parse -q --verify 'sha^74da8e7eee5c214c850a7655c359c39832a64b9e'
      Running command git fetch -q https://github.com/sqlfluff/sqlfluff.git 74da8e7eee5c214c850a7655c359c39832a64b9e
      Running command git checkout -q 74da8e7eee5c214c850a7655c359c39832a64b9e
      error: subprocess-exited-with-error
      
      × Building wheel for sqlfluffrs (pyproject.toml) did not run successfully.
      │ exit code: 1
      ╰─> [98 lines of output]
          Running `maturin pep517 build-wheel -i /Users/sam/.cache/pre-commit/repourh52h32/py_env-python3.11/bin/python --compatibility off`
          🔗 Found pyo3 bindings
          🐍 Found CPython 3.11 at /Users/sam/.cache/pre-commit/repourh52h32/py_env-python3.11/bin/python
          📡 Using build options features from pyproject.toml
          💻 Using `MACOSX_DEPLOYMENT_TARGET=10.12` for x86_64-apple-darwin by default
             Compiling target-lexicon v0.13.4
             Compiling proc-macro2 v1.0.104
             Compiling unicode-ident v1.0.22
             Compiling quote v1.0.42
             Compiling memchr v2.7.6
             Compiling serde_core v1.0.228
             Compiling libc v0.2.178
             Compiling regex-syntax v0.8.8
             Compiling version_check v0.9.5
             Compiling aho-corasick v1.1.4
             Compiling equivalent v1.0.2
             Compiling getrandom v0.3.4
             Compiling serde v1.0.228
             Compiling regex-automata v0.4.13
             Compiling pyo3-build-config v0.26.0
             Compiling syn v2.0.112
             Compiling cfg-if v1.0.4
             Compiling autocfg v1.5.0
             Compiling typenum v1.19.0
             Compiling log v0.4.29
             Compiling foldhash v0.1.5
             Compiling bit-vec v0.8.0
             Compiling allocator-api2 v0.2.21
             Compiling bit-set v0.8.0
             Compiling hashbrown v0.15.5
             Compiling memoffset v0.9.1
             Compiling uuid v1.19.0
             Compiling generic-array v0.14.7
             Compiling heck v0.5.0
             Compiling once_cell v1.21.3
             Compiling pyo3-macros-backend v0.26.0
             Compiling pyo3-ffi v0.26.0
             Compiling pyo3 v0.26.0
             Compiling utf8parse v0.2.2
             Compiling anstyle-parse v0.2.7
             Compiling anstyle-query v1.1.5
             Compiling colorchoice v1.0.4
             Compiling unindent v0.2.4
             Compiling indoc v2.0.7
             Compiling anstyle v1.0.13
             Compiling is_terminal_polyfill v1.70.2
             Compiling anstream v0.6.21
             Compiling itoa v1.0.17
             Compiling either v1.15.0
             Compiling zmij v1.0.8
             Compiling regex v1.12.2
             Compiling fancy-regex v0.16.2
             Compiling hashbrown v0.16.1
             Compiling env_filter v0.1.4
             Compiling serde_derive v1.0.228
             Compiling jiff v0.2.17
             Compiling indexmap v2.12.1
             Compiling itertools v0.14.0
             Compiling block-buffer v0.10.4
             Compiling crypto-common v0.1.7
             Compiling slotmap v1.1.1
             Compiling serde_json v1.0.148
             Compiling unsafe-libyaml v0.2.11
             Compiling subtle v2.6.1
             Compiling ryu v1.0.22
             Compiling digest v0.10.7
             Compiling smallvec v1.15.1
             Compiling blake2 v0.10.6
             Compiling pyo3-macros v0.26.0
             Compiling bincode v1.3.3
             Compiling serde_yaml_ng v0.10.0
             Compiling sqlfluffrs_types v4.0.4 (/private/var/folders/v4/npbx738j76sd232hnyb2r8zc0000gn/T/pip-req-build-6kvf9w3w/sqlfluffrs/sqlfluffrs_types)
             Compiling sqlfluffrs_dialects v4.0.4 (/private/var/folders/v4/npbx738j76sd232hnyb2r8zc0000gn/T/pip-req-build-6kvf9w3w/sqlfluffrs/sqlfluffrs_dialects)
          error[E0583]: file not found for module `dialect`
           --> sqlfluffrs_dialects/src/lib.rs:2:1
            |
          2 | pub mod dialect;
            | ^^^^^^^^^^^^^^^^
            |
            = help: to create the module `dialect`, create file "sqlfluffrs_dialects/src/dialect.rs" or "sqlfluffrs_dialects/src/dialect/mod.rs"
            = note: if there is a `mod dialect` elsewhere in the crate already, import it with `use crate::...` instead
          
          warning: unused import: `dialect::*`
           --> sqlfluffrs_dialects/src/lib.rs:5:9
            |
          5 | pub use dialect::*;
            |         ^^^^^^^^^^
            |
            = note: `#[warn(unused_imports)]` on by default
          
          For more information about this error, try `rustc --explain E0583`.
          warning: `sqlfluffrs_dialects` (lib) generated 1 warning
          error: could not compile `sqlfluffrs_dialects` (lib) due to 1 previous error; 1 warning emitted
          warning: build failed, waiting for other jobs to finish...
          💥 maturin failed
            Caused by: Failed to build a native library through cargo
            Caused by: Cargo build finished with "exit status: 101": `env -u CARGO MACOSX_DEPLOYMENT_TARGET="10.12" PYO3_BUILD_EXTENSION_MODULE="1" PYO3_ENVIRONMENT_SIGNATURE="cpython-3.11-64bit" PYO3_PYTHON="/Users/sam/.cache/pre-commit/repourh52h32/py_env-python3.11/bin/python" PYTHON_SYS_EXECUTABLE="/Users/sam/.cache/pre-commit/repourh52h32/py_env-python3.11/bin/python" "cargo" "rustc" "--profile" "release" "--features" "pyo3/extension-module" "--features" "python" "--message-format" "json-render-diagnostics" "--manifest-path" "/private/var/folders/v4/npbx738j76sd232hnyb2r8zc0000gn/T/pip-req-build-6kvf9w3w/sqlfluffrs/Cargo.toml" "--lib" "--crate-type" "cdylib" "--" "-C" "link-arg=-undefined" "-C" "link-arg=dynamic_lookup" "-C" "link-args=-Wl,-install_name,@rpath/sqlfluffrs.cpython-311-darwin.so"`
          Error: command ['maturin', 'pep517', 'build-wheel', '-i', '/Users/sam/.cache/pre-commit/repourh52h32/py_env-python3.11/bin/python', '--compatibility', 'off'] returned non-zero exit status 1
          [end of output]
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
      ERROR: Failed building wheel for sqlfluffrs
    error: failed-wheel-build-for-install
    
    × Failed to build installable wheels for some pyproject.toml based projects
    ╰─> sqlfluffrs
Check the log at /Users/sam/.cache/pre-commit/pre-commit.log
```

**Comment by keraion** [comment]
Ah, I see the issue here. We don't commit the dialect changes in git so we'd need to have a pre-build step to run to produce these files. I'll need to think of the right way to integrate this properly.

**Comment by keraion** [comment]
Closing this, will track the git-based installation issue in #7576
